packages/taco-auth/src/providers/eip4361/external-eip4361.ts

```diff
@@ -1,18 +1,62 @@
 import { SiweMessage } from 'siwe';
 
+import { AuthProvider } from '../../auth-provider';
+
 import { EIP4361_AUTH_METHOD, EIP4361AuthSignature } from './auth';
 
-export const USER_ADDRESS_PARAM_EXTERNAL_EIP4361 =
-  ':userAddressExternalEIP4361';
+async function generateAndVerifySiweMessage(
+  message: string,
+  signature: string,
+): Promise<SiweMessage> {
+  const siweMessage = new SiweMessage(message);
+  // this will trigger validation for the signature and all parameters (`expirationTime`, `notBefore`...).
+  await siweMessage.verify({ signature });
+  const twoHoursBeforeNow = new Date(Date.now() - 2 * 60 * 60 * 1000);
```

Comment:
Is it possible for us to make the `2 * 60 * 60 * 1000` a constant that could be used by both EIP4361AuthProvider and this one? It would just make maintenance of the value simpler by having it defined in one place, but used in multiple.